packages/usa-form/src/test/test-pattern/test-usa-form.twig

```diff
@@ -16,94 +17,103 @@
   </div>
 </form>
 
-<fieldset class="usa-fieldset margin-top-2">
-  <legend class="usa-legend">Checkbox</legend usa-label>
+<div class="usa-form-group {% if error_state %}usa-form-group--error{% endif %}">
+  <fieldset class="usa-fieldset margin-top-2">
+    <legend class="usa-legend {% if error_state %}usa-label--error{% endif %}">Checkbox{% if error_state %} has an error{% endif %}</legend>
```

Comment:
## Legend and error labels

> [!TIP]
> `usa-legend` and `usa-label` output the same base styles. Because of this, I was able to safely add the `usa-label--error` class without any additional changes or unexpected style conflicts.

Reply:
This seems like an opinionated choice and might cause confusion.

Reply:
Alternatively we could add the `usa-label` class as well but this does not introduce any styles that are not already present thanks to `usa-legend`

Semi-related, we might want to move the duplicate styles from both definitions to a mixin to reduce duplicate code:

https://github.com/uswds/uswds/blob/98566ec5dc12807a2b0bc36699a033fe3a711d44/packages/usa-legend/src/styles/_usa-legend.scss#L3-L13

https://github.com/uswds/uswds/blob/98566ec5dc12807a2b0bc36699a033fe3a711d44/packages/usa-label/src/styles/_usa-label.scss#L3-L13

I've created #5894 for additional discussion and to track work if we decide to move forward with this refactor